esp8266/main.c: Clear the command lines history when (re)booting

robert-hh opened the pull request:
Not clearing the command lines history sometimes resulted in
strange output when going back after reset. If it is intended
to keep the history after soft reset, then the clearing must
be moved e.g. to the start of init_done().

robert-hh committed:
esp8266/main.c: Clear the command lines history when (re)booting

Not clearing the command lines history sometimes resulted in
strange output when going back after reset. If it is intended
to keep the history after soft reset, then the clearing must
be moved e.g. to the start of init_done().

robert-hh commented:
The fails are in the Windows build around machine_pin_base and in some mp_print functions, nothing which has been touched by this PR.  The indicated that there are some inconsistencies in the checking process.

stinos commented:
Appveyor build fails already since https://github.com/micropython/micropython/commit/ad229477c65a5372a4cb77d32f366e7deeaa8099

stinos committed:
windows/msvc: Include machine_pinbase.c in build and enable umachine module

Fixes linker errors since [ad229477] and adds the umachine module so tests pass.

Commit:
extmod/modbtree: Cleverly implement "for key in btree:" syntax.

I.e. make it work like btree.keys(), while still not using a separate
iterator type.

Commit:
tests/btree1: Add testcase for iterating btree object directly.

dpgeorge committed:
lib/libm: Format code to pass gcc v6.1.1 warning.

gcc 6.1.1 warns when indentation is misleading, and in this case the
formatting of the code really is misleading.  So adjust the formatting
to be clear of the meaning of the code.

dpgeorge committed:
lib/libm: Remove unused definition of "one".

robert-hh commented:
Again trouble with git. This time, it included Damien's bug fixes too, which is nice but false honor.
